code/obj/item/implant.dm

```diff
@@ -118,10 +118,13 @@ THROWING DARTS
 					user.show_text("[Imp] already has an implant loaded.")
 					return
 				else
-					user.u_equip(src)
+					var/obj/item/storage/store
+					if(istype(src.loc, /obj/item/storage))
+						store = src.loc
 					src.set_loc(Imp)
 					Imp.imp = src
 					Imp.update()
+					store.hud.remove_item(src)
```

Comment:
This will runtime if it's not in storage

File | implant.dm
-- | --
Line | 127
Error | Cannot read null.hud

```dm
proc name: attackby (/obj/item/implant/attackby)
  source file: implant.dm,127
  usr: Jermaine Young (/mob/living/carbon/human)
  src: the freedom implant (/obj/item/implant/freedom)
  usr.loc: the steel floor (165,180,1) (/turf/simulated/floor)
  src.loc: the implanter (/obj/item/implanter)
  call stack:
the freedom implant (/obj/item/implant/freedom): attackby(the implanter (/obj/item/implanter), Jermaine Young (/mob/living/carbon/human), /list (/list))
Jermaine Young (/mob/living/carbon/human): weapon attack(the freedom implant (/obj/item/implant/freedom), the implanter (/obj/item/implanter), 1, /list (/list))
Jermaine Young (/mob/living/carbon/human): click(the freedom implant (/obj/item/implant/freedom), /list (/list), null, "mapwindow.map")
Jermaine Young (/mob/living/carbon/human): click(the freedom implant (/obj/item/implant/freedom), /list (/list), null, "mapwindow.map")
Sovexe (/client): Click(the freedom implant (/obj/item/implant/freedom), null, "mapwindow.map", "icon-x=17;icon-y=16;left=1;scr...")
```